llvm/lib/TargetParser/TargetDataLayout.cpp

```diff
@@ -275,6 +275,12 @@ static std::string computeAMDDataLayout(const Triple &TT) {
 }
 
 static std::string computeRISCVDataLayout(const Triple &TT, StringRef ABIName) {
+  if (TT.isOSBinFormatMachO()) {
+    assert(TT.isArch32Bit() && "Invalid triple.");
```

Comment:
Big Endian support is slowly landing in LLVM. I presume you want an assertion about that as well.

Reply:
Done, thanks